src/cryptography/hazmat/backends/openssl/x509.py

```diff
@@ -235,6 +238,22 @@ def fingerprint(self, algorithm):
         h.update(der)
         return h.finalize()
 
+    def get_revoked_certificate(self, serial_number):
+        revoked = self._backend._ffi.new("X509_REVOKED **")
+        asn1_int = _encode_asn1_int_gc(self._backend, serial_number)
+        res = self._backend._lib.X509_CRL_get0_by_serial(
```

Comment:
Does this do something more efficient than just a linear search?

Reply:
Implementation here: https://github.com/openssl/openssl/blob/01ad66f85d22fd001582b5f2e6e18db8b820c550/crypto/x509/x_crl.c#L381-L414

Short answer: not really. It sorts and then does a linear asn1 integer comparison. But no copies across ffi boundary.

Reply:
Ah, actually the stack find uses a binary search once it sorts the list. So it is faster than a linear search (in most cases)